src/coreclr/jit/target.h

```diff
@@ -305,56 +364,69 @@ struct regMaskTP
 
     SingleTypeRegSet GetPredicateRegSet() const
     {
+#ifdef HAS_MORE_THAN_64_REGISTERS
+        return getHigh();
+#else
         return getLow();
+#endif
     }
 
-    void RemoveRegNumFromMask(regNumber reg);
+    void operator|=(const regMaskTP& second)
+    {
+        low |= second.getLow();
+#ifdef HAS_MORE_THAN_64_REGISTERS
+        high |= second.getHigh();
+#endif
+    }
 
-    bool IsRegNumInMask(regNumber reg);
+    void operator^=(const regMaskTP& second)
+    {
+        low ^= second.getLow();
+#ifdef HAS_MORE_THAN_64_REGISTERS
+        high ^= second.getHigh();
+#endif
+    }
+
+    void operator&=(const regMaskTP& second)
+    {
+        low &= second.getLow();
+#ifdef HAS_MORE_THAN_64_REGISTERS
+        high &= second.getHigh();
+#endif
+    }
+
+    void operator|=(const regNumber reg);
+    void operator^=(const regNumber reg);
```

Comment:
It would be nice to avoid having these operators (they seem odd to me). Can we just use the `AddRegNum/RemoveRegNum` funcs instead?

Reply:
can't agree more. I brought this from my original PR, but it doesn't make sense to just use explicit methods for these operations.